src/audio/crossover/crossover.c

```diff
@@ -106,12 +107,24 @@ static int crossover_get_stream_index(struct sof_crossover_config *config,
 		if (assign_sink[i] == pipe_id)
 			return i;
 
-	comp_cl_err(&comp_crossover, "crossover_get_stream_index() error: couldn't find any assignment for connected pipeline %u",
-		    pipe_id);
+	comp_err(mod->dev, "crossover_get_stream_index() error: couldn't find any assignment for connected pipeline %u",
+		 pipe_id);
 
 	return -EINVAL;
 }
 
+static uint32_t crossover_get_sink_pipeline_id(struct comp_buffer __sparse_cache *sink_c)
+{
+#if CONFIG_IPC_MAJOR_4
+	// TODO: IPC4 renumbers pipeline_id:s due to pipelines count limitation, so can't use it.
+	// Use instead output pin index, cd->ipc4_cfg->output_pin.pin_index. The crossover
+	// topology blobs for IPC4 are now built with those indices.
+	return sink_c->sink->pipeline->pipeline_id;
```

Comment:
This change is stll needed. The crossover configuration blobs for IPC4 need to be populated with pin indexes 0, 1, ... instead of pipeline_id:s.